hazelcast/src/main/java/com/hazelcast/internal/ascii/rest/HttpPostCommandProcessor.java

```diff
@@ -890,4 +894,31 @@ private void sendResponse(HttpPostCommand command, String value) {
     public void handleRejection(HttpPostCommand command) {
         handle(command);
     }
+
+    private void handleSetLicense(HttpPostCommand command) {
+        final int retryCount = 100;
+        String res;
+        byte[] data = command.getData();
+        try {
+            String[] strList = bytesToString(data).split("&");
+            if (authenticate(command, strList[0], strList.length > 1 ? strList[1] : null)) {
+                // assumes that both groupName and password are present
+                String licenseKey = strList.length > 2 ? URLDecoder.decode(strList[2], "UTF-8") : null;
+                invokeOnStableClusterSerial(textCommandService.getNode().nodeEngine,
+                        () -> new SetLicenseOperation(licenseKey), retryCount).get();
```

Comment:
In 3.12.z stream we need to avoid lambdas to keep the possibility to feed users with Java 6 compatible binaries.
```
$ mvn clean install -P-jdk-8 -U -B -DskipTests
...
[INFO] --- maven-compiler-plugin:2.5.1:compile (default-compile) @ hazelcast ---
[INFO] Compiling 4316 source files to /home/kwart/projects/hazelcast/hazelcast/hazelcast/target/classes
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /home/kwart/projects/hazelcast/hazelcast/hazelcast/src/main/java/com/hazelcast/internal/ascii/rest/HttpPostCommandProcessor.java:[908,27] error: lambda expressions are not supported in -source 1.6
[INFO] 1 error
...
```

Reply:
Ah, many thanks, good catch. Should be fixed now.